cdap-ui/app/features/hydratorplusplus/controllers/create/partials/postactions-ctrl.js

```diff
@@ -0,0 +1,191 @@
+    this.HydratorPlusPlusConfigActions = HydratorPlusPlusConfigActions;
+    this.myAlertOnValium = myAlertOnValium;
+
+    this.HydratorPlusPlusConfigStore.registerOnChangeListener(this.setState.bind(this));
+
+    let artifact = this.HydratorPlusPlusConfigStore.getArtifact();
+    this.postActionsList = [];
+
+    let params = {
+      namespace: this.$state.params.namespace,
+      pipelineType: artifact.name,
+      version: artifact.version,
+      extensionType: 'postaction'
+    };
+    this.myPipelineApi.fetchPlugins(params)
+      .$promise
+      .then( (res) => {
+        this.postActionsList = res;
+      });
+
+    this.setState();
+
+  }
+  setState() {
+    this.postactions = this.HydratorPlusPlusConfigStore.getPostActions();
+    this.activeTab = 'OVERVIEW';
+  }
+
+  tabClick(tab) {
+    this.activeTab = tab;
+
+    if (tab === 'OVERVIEW') { return; }
+
+    this.isEdit = false;
+    this.pluginConfig = {
+      name: tab.name + '-' + this.uuid.v4(),
+      plugin: {
+        name: tab.name,
+        type: tab.type,
+        artifact: tab.artifact,
+        properties: {}
+      }
+    };
+
+    this.pluginFetch(tab);
+  }
+
+  edit(config) {
+    this.isEdit = true;
+    let tab = {
+      artifact: {
+        name: config.plugin.artifact.name,
+        version: config.plugin.artifact.version,
+        scope: config.plugin.artifact.scope
+      },
+      name: config.plugin.name,
+      type: config.plugin.type
+    };
+
+    this.pluginConfig = config;
+    this.pluginFetch(tab);
+
+    this.activeTab = tab;
+  }
+
+  // Fetching Backend Properties
+  pluginFetch(tab) {
+    let params = {
+      namespace: this.$state.params.namespace,
+      pipelineType: tab.artifact.name,
+      version: tab.artifact.version,
+      scope: tab.artifact.scope,
+      extensionType: tab.type,
+      pluginName: tab.name
+    };
+
+    this.myPipelineApi.fetchPluginProperties(params)
+      .$promise
+      .then( (res) => {
+        this.pluginConfig._backendProperties = res[0].properties;
+        this.fetchWidgets(tab);
+      });
+  }
+
+  // Fetching Widget JSON for the plugin
+  fetchWidgets(tab) {
+    let artifact = {
+      name: tab.artifact.name,
+      version: tab.artifact.version,
+      scope: tab.artifact.scope,
+      key: 'widgets.' + tab.name + '-' + tab.type
+    };
+    this.HydratorPlusPlusPluginConfigFactory
+      .fetchWidgetJson(artifact.name, artifact.version, artifact.scope, artifact.key)
+      .then( (widgetJson) => {
+        this.noConfig = false;
+
+        this.groupsConfig = this.HydratorPlusPlusPluginConfigFactory.generateNodeConfig(this.pluginConfig._backendProperties, widgetJson);
+
+        // Initializing default value
+        angular.forEach(this.groupsConfig.groups, (group) => {
+          angular.forEach(group.fields, (field) => {
+            if (field.defaultValue) {
+              this.pluginConfig.properties[field.name] = this.pluginConfig.properties[field.name] || field.defaultValue;
```

Comment:
Oh, that's weird. Alright.
On Apr 15, 2016 11:54 AM, "Edwin Elia" <notifications@github.com> wrote:

> In
> cdap-ui/app/features/hydratorplusplus/controllers/create/partials/postactions-ctrl.js
> <https://github.com/caskdata/cdap/pull/5547#discussion_r59923439>:
>
> > +      version: tab.artifact.version,
> > +      scope: tab.artifact.scope,
> > +      key: 'widgets.' + tab.name + '-' + tab.type
> > +    };
> > +    this.HydratorPlusPlusPluginConfigFactory
> > +      .fetchWidgetJson(artifact.name, artifact.version, artifact.scope, artifact.key)
> > +      .then( (widgetJson) => {
> > +        this.noConfig = false;
> > +
> > +        this.groupsConfig = this.HydratorPlusPlusPluginConfigFactory.generateNodeConfig(this.pluginConfig._backendProperties, widgetJson);
> > +
> > +        // Initializing default value
> > +        angular.forEach(this.groupsConfig.groups, (group) => {
> > +          angular.forEach(group.fields, (field) => {
> > +            if (field.defaultValue) {
> > +              this.pluginConfig.properties[field.name] = this.pluginConfig.properties[field.name] || field.defaultValue;
>
> that syntax is actually really unreliable after testing. There are cases
> where it still go to the defaultValue even though the field has value
> already
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/caskdata/cdap/pull/5547/files/27e78e5fc41955b8e5ffc3e9ca40c765c2f920b2#r59923439>
>